### PR TITLE
Tab bar

### PR DESCRIPTION
**Pull request by gastonmaspero**
## Summary ##

**Trello card**: https://trello.com/c/vYIIemhW/9-n-43-tab-bar

Description: Added tab bar

### iPhone X

Simulator Screen Shot - iPhone X - 2019-05-27 at 13 36 11

### CODE REVIEW

File: WBooks/MainTabBar/LibraryTab.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTabBar()
+        addNavBarButtons()
+    }
+    
+    private func addNavBarButtons() {
+        self.title = "NAVIGATION_BAR_TITLE".localized()
+        self.setNavigationBarTitle("NAVIGATION_BAR_TITLE".localized(), font: UIFont.systemFont(ofSize: 20, weight: UIFont.Weight.medium), color: UIColor.white)
```
Comment (carcos-wolox) [inline WBooks/MainTabBar/LibraryTab.swift]: unnecessary `self`

File: WBooks/MainTabBar/LibraryTab.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        let rentals = RentalsController()
+        rentals.tabBarItem = UITabBarItem()
+        rentals.tabBarItem.title = "Rentals"
+        rentals.tabBarItem.image = UIImage(named: "ic_myrentals")
+        rentals.tabBarItem.tag = 3
+        
+        let settings = SettingsController()
+        settings.tabBarItem = UITabBarItem()
+        settings.tabBarItem.title = "Settings"
+        settings.tabBarItem.image = UIImage(named: "ic_settings")
```
Comment (carcos-wolox) [inline WBooks/MainTabBar/LibraryTab.swift]: We use to have an `UIImageExtension` to keep all this Images in the extension and use them as what you did with UIBarButtonItem. So, create an extension of UIImage and set there the images like:
`static var settingsImage: UIImage {
        return UIImage(named: "ic_settings"))
  }`

File: WBooks/MainTabBar/LibraryTab.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        navigationItem.leftBarButtonItem = UIBarButtonItem.notificationButton()
+        //Right search button
+        navigationItem.rightBarButtonItem = UIBarButtonItem.searchButton()
+    }
+    
+    private func configureTabBar() {
+        tabBar.barTintColor = .white
+        let library = MainMenuController()
+        library.tabBarItem = UITabBarItem()
+        library.tabBarItem.title = "Library"
```
Comment (carcos-wolox) [inline WBooks/MainTabBar/LibraryTab.swift:37]: This strings should be on the localized file that you've created in the other card.

File: WBooks/Library/MainMenuController.swift [hunk 4]
```diff
@@ -34,24 +35,12 @@ class MainMenuController: UIViewController {
...
-        //Right search button
-        navigationItem.rightBarButtonItem = UIBarButtonItem.searchButton()
-    }
-    
     private func configureTableView() {
         books = createArray()
         _view.tableView.delegate = self
         _view.tableView.dataSource = self
         _view.tableView.register(cell: BookCell.self)
-        _view.tableView.rowHeight = UITableViewAutomaticDimension
```
Comment (carcos-wolox) [inline WBooks/Library/MainMenuController.swift:52]: Why you delete this?

File: WBooks/MainTabBar/LibraryTab.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+class LibraryTab: UITabBarController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTabBar()
+        addNavBarButtons()
+    }
+    
+    private func addNavBarButtons() {
```
Comment (carcos-wolox) [inline WBooks/MainTabBar/LibraryTab.swift:24]: You also need to set the status bar color to white

**Review by carcos-wolox — CHANGES_REQUESTED** [review]
You are going an excellent job!! I've added some small comments

File: WBooks/MainTabBar/LibraryTab.swift (hunk 5 above)
Comment (gastonmaspero) [reply WBooks/MainTabBar/LibraryTab.swift:24]: The method `configureTabBar()` starts with `tabBar.barTintColor = .white`, which does that.

File: WBooks/Library/MainMenuController.swift (hunk 4 above)
Comment (gastonmaspero) [reply WBooks/Library/MainMenuController.swift:52]: I have this code:
`func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 100
    }`

Was that statement necessary? I checked and no changes came up on deletion.

Comment (carcos-wolox) [reply WBooks/Library/MainMenuController.swift:52]: If you set `rowHeight` and `estimatedRowHeight` you don't need that function. But it's ok 👌

File: WBooks/MainTabBar/LibraryTab.swift (hunk 5 above)
Comment (carcos-wolox) [reply WBooks/MainTabBar/LibraryTab.swift:24]: I mean, the bar that is over to the navigationBar.
image

**Review by carcos-wolox — APPROVED** [review]
Excellent! Remember to update the screenshot in the description with the light status bar

**Pull request merged**